src/LtiMessageLaunch.php

```diff
@@ -118,52 +120,27 @@ public static function fromCache(
         ICache $cache = null,
         ILtiServiceConnector $serviceConnector = null
     ) {
+        // @TODO: Fix the null here on the next major version
         $new = new LtiMessageLaunch($database, $cache, null, $serviceConnector);
         $new->launch_id = $launch_id;
         $new->jwt = ['body' => $new->cache->getLaunchData($launch_id)];
 
         return $new->validateRegistration();
     }
 
-    public function setRequest(array $request = null)
+    public function setRequest(array $request)
     {
-        if ($request === null) {
-            $request = $_POST;
-        }
-
         $this->request = $request;
 
         return $this;
     }
 
-    private function shouldMigrate(): bool
-    {
-        assert($this->db instanceof IMigrationDatabase);
-
-        $launchData = $this->getLaunchData();
-        $lti1p1Install = $this->db->getMatchingLti1p1Install($launchData);
-        if ($lti1p1Install !== null) {
-            return $this->lti1p1InstallationSignatureMatches($lti1p1Install, $launchData);
-        }
-
-        return false;
-    }
-
-    private function lti1p1InstallationSignatureMatches(Lti1p1Installation $ltiInstall, array $launchData): bool
+    public function initialize(array $request)
     {
-        // Check to see if we have params to calculate oauth_consumer_key_sign
-        if (!isset($launchData[LtiConstants::LTI1P1]) || !isset($launchData[LtiConstants::LTI1P1]['oauth_consumer_key_sign'])) {
-            throw new LtiException(static::ERR_MISSING_OAUTH_CONSUMER_KEY_SIGN);
-        }
-
-        $lti1p1Claim = $launchData[LtiConstants::LTI1P1];
-        $signature = $lti1p1Claim['oauth_consumer_key_sign'];
-
-        if (!$ltiInstall->compareOauthConsumerKeySign($launchData, $signature)) {
-            throw new LtiException(static::ERR_OAUTH_CONSUMER_KEY_SIGN_MISMATCH);
-        }
-
-        return true;
+        return $this->setRequest($request)
+            ->validate()
+            ->migrate()
+            ->cacheLaunchData();
```

Comment:
Doesn't `validate` already `cacheLaunchData()`? @dbhynds ?

Reply:
It does, but I want to remove it from there in the next major version. IMO people should either do:
```php
$launch = LtiMessageLaunch::new()->initialize($request);

// Or manually by doing this (if they wanted to omit migrations or some other step):
$launch = LtiMessageLaunch::new()->setRequest($request)
    ->validate()
    ->cacheLaunchData();
```

Reply:
That sounds fine then